test/jdk/javax/swing/JFileChooser/FileChooserSymLinkTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+/* @test
+   @bug 8281966
+   @key headful
+   @requires (os.family == "windows")
+   @library /java/awt/regtesthelpers
+   @build PassFailJFrame
+   @summary Test to check if the absolute path of Symbolic Link folder
+            is valid on ValueChanged property listener.
+   @run main FileChooserSymLinkTest
+*/
+
+import java.awt.Dimension;
+import java.awt.Robot;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+
+public class FileChooserSymLinkTest{
+    static JFrame frame = null;
+    static PassFailJFrame passFailJFrame = null;
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                try {
+                    initialize();
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                } catch (InvocationTargetException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+
+        robot.delay(2000);
+        robot.waitForIdle();
+
+        passFailJFrame.awaitAndCheck();
+    }
+
+    static void initialize() throws InterruptedException, InvocationTargetException {
+        final String INSTRUCTIONS = """
+                Instructions to Test:
+                1. Create a regular directory in any specific path.
+                    ex: mkdir c:\\target
+                2. Create a Symbolic link targeting the created test directory.
+                    ex : mklink /D c:\\link c:\\target
+                3. In JFileChooser, navigate to "link" created directed.
```

Comment:
…navigate to directory where you created the link. In this example, to C:\ drive.

Please use the capital C: for the drive.

Reply:
Updated.

Reply:
I can't see it updated.

In fact I suggest providing a script.

```
1. Open an elevated Command Prompt.
2. Paste the following commands:

cd /d C:\
mkdir filechooser
cd filechooser
mkdir target
mklink /d link target

3. Navigate to C:\filechooser in the JFileChooser.
4. ...
5. When done with testing, paste the following commands to remove the 'filechooser' directory

cd \
rmdir /s /q filechooser

or use File Explorer to clean it up.
```